src/onnxruntime.cc

```diff
@@ -872,6 +872,7 @@ ModelState::LoadModel(
          "' on device " + std::to_string(instance_group_device_id) +
          std::string(" with options: ") + std::string(options))
             .c_str());
+    RETURN_IF_ORT_ERROR(ort_api->AllocatorFree(allocator, options));
```

Comment:
Should double check if this is okay
> * \param[out] out Returned value should NOT be freed
https://github.com/microsoft/onnxruntime/blob/281ed8c12d2d2a3f5b683e6267aa0fca4d4add50/include/onnxruntime/core/session/onnxruntime_c_api.h#L1641